overlord.pl

```diff
@@ -342,13 +350,35 @@ sub finishCaching
     &waitFor("swapouts gone", sub { ! &squidHasSwapouts() });
 }
 
+sub waitActiveRequests
+{
+    my ($options) = @_;
+
+    my $requestPath = $options->{$onameRequestPath};
+    die("missing $onameRequestPath\n") unless defined $requestPath;
+    my $activeRequestsCount = $options->{$onameActiveRequestsCount};
+    die("missing $onameActiveRequestsCount\n") unless defined $activeRequestsCount;
+
+    my ($path, $count) = @_;
+    &waitFor("until all $activeRequestsCount requests become active", sub { &squidActiveRequests($requestPath, $activeRequestsCount) });
+}
+
 # whether Squid has StoreEntries in SWAPOUT_WRITING state
 sub squidHasSwapouts
 {
     my $mgrPage = &getCacheManagerResponse('openfd_objects')->{content};
     return $mgrPage =~ /SWAPOUT_WRITING/;
 }
 
+sub squidActiveRequests
+{
+    my $path = shift;
+    my $count = shift;
+    my $mgrPage = &getCacheManagerResponse('active_requests')->{content};
+    my @matches = $mgrPage =~ /$path/g;
+    return scalar @matches >= $count;
```

Comment:
Please refactor a little so that this function returns the number of matching active requests. The caller should check whether that number meets caller's needs. It would be easier to name and reuse this function that way.

Please call the adjusted function `countMatchingActiveRequests`.

```suggestion
    return scalar @matches;
```

Reply:
Ok.